src/astarte-client/client.ts

```diff
@@ -407,25 +410,38 @@ class AstarteClient {
     await this.$delete(this.apiConfig.flowInstance({ ...this.config, instanceName: flowName }));
   }
 
-  async getPipelineDefinitions(): Promise<any> {
+  async getPipelineNames(): Promise<Array<AstartePipeline['name']>> {
     const response = await this.$get(this.apiConfig.pipelines(this.config));
     return response.data;
   }
 
-  async registerPipeline(pipeline: any): Promise<void> {
-    await this.$post(this.apiConfig.pipelines(this.config), pipeline);
+  async getPipelines(): Promise<AstartePipeline[]> {
+    const pipelineNames = await this.getPipelineNames();
+    const response = await Promise.allSettled(pipelineNames.map(this.getPipeline));
+    const pipelines: AstartePipeline[] = [];
+    response.forEach((r) => {
+      if (r.status === 'fulfilled') {
```

Comment:
checking a promise status is an indicator of a bad pattern.
You can use `Promise.all` to make this much cleaner
```
pipelineNames = getNames()
promiseArray = pipelineNames.map( name => getDetail(name) )
return Promise.all( promiseArray ).then( results => results.map( pipelineDetails => parse(pipelineDetails) ) )
```
```

Reply:
Indeed, it is better not to discard errors and pass them on to the React view. I changed from Promise.allSettled to Promise.all